app/web/features/auth/useAuthStore.ts

```diff
@@ -131,12 +135,53 @@ export default function useAuthStore() {
         }
         setLoading(false);
       },
+      async checkAuthStatus() {
+        try {
```

Comment:
I wonder if adding a check `if (authenticated || loading) return;` would help here instead of the useEffect? That's kinda what you're trying to do yeah?

Reply:
Let me comment here instead of the other two.

`checkAuthStatus` is right now only run if `isAuthenticated` is true, but I've left that condition in the `useEffect`. This is because of the comment at the bottom "there should be no dependenices on the state". I originally had it there, but then read this...

Also, regarding logging out: there is a well functioning mechanism that will log the user out in the frontend if the backend returns an unauthenticated response via the API.